### PR TITLE
Clean up the gulp tasks for documentation

### PR DESCRIPTION
**Pull request by andy-armstrong**
## Description

https://openedx.atlassian.net/browse/FEDX-148

This change adopts a number of the lessons learned from creating the UI Toolkit documentation:
 - move all the Jekyll content beneath the pldoc directory
 - gitignore the public directories so that they don't get committed
 - introduce a new gulp doc-publish task to simplify updating GitHub Pages
 - changed preview so that it opens the S3 URL that it builds

Here's an updated preview site just to verify that things are still working:

http://ux-test.edx.org/andya/improve-github-pages-integration/ 

## Reviewers

 - [x] @bjacobel 
 - [ ] @AlasdairSwan

### CODE REVIEW

File: gulp/tasks/fonts.js [hunk 1]
```diff
@@ -7,12 +9,11 @@ var gulp            = require('gulp'),
 gulp.task('fonts', function() {
 
     var fonts = gulp.src(config.src_files);
-    var pldoc_fonts = gulp.src(config.pldoc_src_files);
-    var demo_fonts = gulp.src(config.demo_src_files);
+    var pldocFonts = gulp.src(config.pldoc_src_files);
+    var demoFonts = gulp.src(config.demo_src_files);
```
Comment (AlasdairSwan) [inline gulp/tasks/fonts.js]: There should only be 1 ```var```, comma separating the variables.

File: gulp/tasks/images.js [hunk 2]
```diff
@@ -9,16 +11,15 @@ var gulp            = require('gulp'),
 gulp.task('images', function() {
 
     var images = gulp.src(config.src_files);
-    var pldoc_images = gulp.src(config.pldoc_src_files);
-    var demo_images = gulp.src(config.demo_src_files);
+    var pldocImages = gulp.src(config.pldoc_src_files);
+    var demoImages = gulp.src(config.demo_src_files);
```
Comment (AlasdairSwan) [inline gulp/tasks/images.js]: Nitpick: there should only be 1 ```var```, comma separating the variables.

**Comment by AlasdairSwan** [comment]
Just a few nitpicks, looks good @andy-armstrong 👍

**Comment by bjacobel** [comment]
I'm unable to verify this locally because of https://openedx.atlassian.net/browse/FEDX-178.

**Comment by andy-armstrong** [comment]
@bjacobel @AlasdairSwan I've rebased to pick up the fixes to the gulp pipeline, and I've fixed the nitpicks.

**Comment by bjacobel** [comment]
Getting the following error running locally:

```
$ > node_modules/.bin/gulp
[16:13:34] Using gulpfile ~/code/ux-pattern-library/gulpfile.js
[16:13:34] Starting 'default'...
[16:13:34] Starting 'build-development'...
[16:13:34] Starting 'clean'...
[16:13:34] Finished 'clean' after 7.84 ms
[16:13:34] Starting 'fonts'...
[16:13:34] Finished 'fonts' after 124 ms
[16:13:34] Starting 'images'...
[16:13:34] 'images' errored after 1.53 ms
[16:13:34] Error: Invalid output folder
    at Gulp.dest (/Users/bjacobel/code/ux-pattern-library/node_modules/vinyl-fs/lib/dest/index.js:15:11)
    at Gulp.<anonymous> (/Users/bjacobel/code/ux-pattern-library/gulp/tasks/images.js:17:20)
    at module.exports (/Users/bjacobel/code/ux-pattern-library/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/bjacobel/code/ux-pattern-library/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/bjacobel/code/ux-pattern-library/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/Users/bjacobel/code/ux-pattern-library/node_modules/orchestrator/index.js:134:8)
    at runNextSet (/Users/bjacobel/code/ux-pattern-library/node_modules/run-sequence/index.js:88:16)
    at Gulp.onTaskEnd (/Users/bjacobel/code/ux-pattern-library/node_modules/run-sequence/index.js:77:5)
    at emitOne (events.js:95:20)
    at Gulp.emit (events.js:182:7)
[16:13:34] 'default' errored after 137 ms
[16:13:34] Error in plugin 'run-sequence'
Message:
    An error occured in task 'images'.
[16:13:34] 'build-development' errored after 137 ms
[16:13:34] Error in plugin 'run-sequence'
Message:
    An error occured in task 'default'.
[16:13:34] 'build-development' errored after 137 ms
[16:13:34] Error in plugin 'run-sequence'
Message:
    An error occured in task 'images'.
```

The references to `'images'` would initially make me thing this was introduced in #325, but this does not occur on master.

(Node 5.9.1)

**Comment by andy-armstrong** [comment]
@bjacobel Bad merge, sorry. Please try again...

**Comment by bjacobel** [comment]
Looks good to me, thumbed.

**Pull request merged**